crates/ruff_linter/src/rules/pyflakes/rules/strings.rs

```diff
@@ -888,3 +943,12 @@ pub(crate) fn string_dot_format_mixing_automatic(
         checker.report_diagnostic(StringDotFormatMixingAutomatic, call.range());
     }
 }
+
+/// Searches for a call expression by looking for a [`TokenKind::Lpar`] token.
+fn contains_call_expr_in_range(checker: &Checker, range: TextRange) -> bool {
```

Comment:
Using `contains_effect`  would be more reliable and would catch other kinds of expression with side effects.

Reply:
You can call it like this

```
if !contains_effect(target, |id| checker.semantic().has_builtin_binding(id)) {
```

`has_builtin` is used to find some builtins that are known to have no side effect. Removing calls without a side effect is fine because they have no effect.